src/main/java/org/jabref/model/util/FileHelper.java

```diff
@@ -104,17 +123,44 @@ public static Optional<Path> find(String fileName, List<Path> directories) {
                           .flatMap(directory -> find(fileName, directory).stream())
                           .findFirst();
     }
+    /**
+     * Detect illegal characters in given filename.
+     *
+     * @param fileName the fileName to detect
+     * @return Boolean whether there is a illegal name.
+     *
+     */
+
+    public static boolean detectBadFileName(String fileName) {
+        for (int i = 0; i < fileName.length(); i++) {
+            char c = fileName.charAt(i);
+            if (!isCharLegal(c)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    public static boolean isCharLegal(char c) {
+        return Arrays.binarySearch(ILLEGAL_CHARS, c) < 0;
+    }
 
     /**
      * Converts a relative filename to an absolute one, if necessary. Returns
      * an empty optional if the file does not exist.
+     *
      */
+
     public static Optional<Path> find(String fileName, Path directory) {
         Objects.requireNonNull(fileName);
         Objects.requireNonNull(directory);
-
         // Explicitly check for an empty String, as File.exists returns true on that empty path, because it maps to the default jar location
         // if we then call toAbsoluteDir, it would always return the jar-location folder. This is not what we want here
+        if (detectBadFileName(fileName)) {
+            LOGGER.error("Invalid characters in path");
```

Comment:
```suggestion
            LOGGER.error("Invalid characters in path for file {} ", fileName);
```
This gives the user context in the erorr log